dtls.c

```diff
@@ -969,54 +951,6 @@ int janus_dtls_verify_callback(int preverify_ok, X509_STORE_CTX *ctx) {
 	return 1;
 }
 
-/* DTLS BIOs to/from socket bridge */
-void janus_dtls_fd_bridge(janus_dtls_srtp *dtls) {
-	if(dtls == NULL) {
-		JANUS_LOG(LOG_ERR, "No DTLS-SRTP stack, no DTLS bridge...\n");
-		return;
-	}
-	janus_ice_component *component = (janus_ice_component *)dtls->component;
-	if(component == NULL) {
-		JANUS_LOG(LOG_ERR, "No component, no DTLS bridge...\n");
-		return;
-	}
-	janus_ice_stream *stream = component->stream;
-	if(!stream) {
-		JANUS_LOG(LOG_ERR, "No stream, no DTLS bridge...\n");
-		return;
-	}
-	janus_ice_handle *handle = stream->handle;
-	if(!handle || !handle->agent || !dtls->write_bio) {
-		JANUS_LOG(LOG_ERR, "No handle/agent/bio, no DTLS bridge...\n");
-		return;
-	}
-	int pending = BIO_ctrl_pending(dtls->filter_bio);
-	while(pending > 0) {
-		JANUS_LOG(LOG_HUGE, "[%"SCNu64"] >> Going to send DTLS data: %d bytes\n", handle->handle_id, pending);
-		char outgoing[pending];
-		int out = BIO_read(dtls->write_bio, outgoing, sizeof(outgoing));
-		JANUS_LOG(LOG_HUGE, "[%"SCNu64"] >> >> Read %d bytes from the write_BIO...\n", handle->handle_id, out);
-		if(out > 1500) {
-			/* FIXME Just a warning for now, this will need to be solved with proper fragmentation */
-			JANUS_LOG(LOG_WARN, "[%"SCNu64"] The DTLS stack is trying to send a packet of %d bytes, this may be larger than the MTU and get dropped!\n", handle->handle_id, out);
```

Comment:
I've noticed that you've removed this warning where you now send the packet, could you add it again? Maybe right after the "negative size" check, as that's where we're checking the size already anyway.

Reply:
The handle is not known yet at the top of the new function, so I re-added the warning just before the call to `nice_agent_send`. Shouldn't the check be against the configured `mtu` anyway? If somebody configures the MTU to be 2000 bytes, the warning should not be shown imho.

Also if you want to avoid sending packets larger than 1500 bytes, it should work to return `min(mtu, 1500)` from the `BIO_CTRL_DGRAM_QUERY_MTU` in `janus_dtls_bio_agent_ctrl`.

Reply:
I don't really want to prevent that: MTU is only a problem when it is. In case Janus is used entirely in a local network it might not be a problem at all. But it's a good point that we should add the warning when setting the MTU in the first place.